pandas/_libs/groupby.pyx

```diff
@@ -778,6 +778,10 @@ def group_quantile(ndarray[float64_t] out,
             if not mask[i]:
                 non_na_counts[lab] += 1
 
+    if labels.any():
+        # Put '-1' (NaN) labels as the last group so it does not interfere
+        # with the calculations.
+        labels[labels==-1] = np.max(labels) + 1
```

Comment:
Small nit but can you put spaces around the ==? black won't catch this because it's in a pyx file